src/components/website-management/website-form-components/customization/customization-components/BannerTemplate.js

```diff
@@ -0,0 +1,115 @@
+import React, {useContext, useEffect, useState} from "react";
+import {Box} from "@mui/system";
+import Typography from "@mui/material/Typography";
+import CustSelectBox from "./CustSelectBox";
+import {Button, TextField} from "@mui/material";
+import axios from "axios";
+import PopUp from "./PopUp";
+import {BannerContext} from "./BannerContext";
+
+
+const BannerTemplate = () =>{
+
+    const[openPopup,setOpenPopup] = useState(false);
+
+    const [selectedTemplate, setSelectedTemplate] = useState({});
+
+    const [templates,settemplates] = useState([]);
+
+    const passedId = useContext(BannerContext).websiteId;
+
+    console.log("passed id temp"+passedId);
+
+
+   const loadTemplates = async () => {
+
+        const websiteID = passedId;
+        const result = await axios.get(`http://localhost:8080/api/template/getAllTemplates/${websiteID}`);
+        console.log("getAllTemplates :", result.data);
+        settemplates(result.data);
+
+        console.log("selectedTemplate length", Object.keys(selectedTemplate).length)
+        console.log("selectedTemplate length", selectedTemplate)
+
+        if(Object.keys(selectedTemplate).length !== 0){
+            const template = result.data.find((t) => t.templateRegulation === selectedTemplate?.templateRegulation);
+            setSelectedTemplate(template);
+        } else {
+            const template = result.data[0];
+            console.log("Empty Value : ", template);
+            setSelectedTemplate(template);
+        }
+    }
+    useEffect(() => {
+        loadTemplates();
+      /*  console.log("this is temps"+templates);*/
+    }, []);
+
+
+    const editButtonAction = () => {
+        setOpenPopup(true);
+    }
+
+    const handleClose = async () => {
+        setOpenPopup(false);
+        // console.log("handleClose : ",selectedTemplate?.templateRegulation)
+        // const template = templates.find((t) => t.templateRegulation === selectedTemplate?.templateRegulation);
+        // console.log("templates : ",template)
+        // setSelectedTemplate(template);
+    }
+
+
+    const regulationValues = templates?.length > 0 ? templates.map(template => ({ id: template.templateId, value: template.templateRegulation }))
+        : [];
+
+   /* console.log("this is regu", JSON.stringify(regulationValues));
+    console.log("this is zero", JSON.stringify(regulationValues[0].value));*/
+
+    const choosenTemplate = (e) => {
+        /*console.log("Chose Temp : ", e.target.value)*/
+        const template = templates.find((t) => t.templateRegulation === e.target.value);
+        setSelectedTemplate(template);
+
+    }
+
+
+
+
+    return(
+        <div className={"TemplateView"}>
```

Comment:
Let's format jsx properly. Use a tool like prettier for that.

Reply:
fixed